crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -0,0 +1,405 @@
+	};
+	let contract_artifacts =
+		ContractArtifacts::from_manifest_or_file(Some(&cargo_toml_path), None)?;
+	let transcoder = contract_artifacts.contract_transcoder()?;
+	let mut constructors: Vec<Constructor> = Vec::new();
+	for constructor in transcoder.metadata().spec().constructors() {
+		constructors.push(Constructor {
+			label: constructor.label().to_string(),
+			payable: *constructor.payable(),
+			args: process_args(constructor.args()),
+			docs: constructor.docs().join(" "),
+			default: *constructor.default(),
+		});
+	}
+	Ok(constructors)
+}
+
+/// Extracts the information of a smart contract constructor parsing the metadata file.
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+/// * `constructor` - The label of the constructor.
+fn get_constructor<P>(path: P, constructor: &str) -> Result<Constructor, Error>
+where
+	P: AsRef<Path>,
+{
+	get_constructors(path.as_ref())?
+		.into_iter()
+		.find(|c: &Constructor| c.label == constructor)
+		.ok_or_else(|| Error::InvalidConstructorName(constructor.to_string()))
+}
+
+// Parse the parameters into a vector of argument labels.
+fn process_args(params: &[MessageParamSpec<PortableForm>]) -> Vec<Param> {
+	let mut args: Vec<Param> = Vec::new();
+	for arg in params {
+		args.push(Param {
+			label: arg.label().to_string(),
+			type_name: arg.ty().display_name().to_string(),
+		});
+	}
+	args
+}
+
+/// Processes a list of argument values for a specified contract message,
+/// wrapping each value in `Some(...)` or replacing it with `None` if the argument is optional
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+/// * `message_label` - Label of the contract message to retrieve.
+/// * `args` - Argument values provided by the user.
+pub fn process_message_args<P>(
+	path: P,
+	message: &str,
+	args: Vec<String>,
+) -> Result<Vec<String>, Error>
+where
+	P: AsRef<Path>,
+{
+	let message = get_message(path, message)?;
+	if args.len() != message.args.len() {
+		return Err(Error::IncorrectArguments {
+			expected: message.args.len(),
+			provided: args.len(),
+		});
+	}
+	Ok(args
+		.into_iter()
+		.zip(&message.args)
+		.map(|(arg, param)| match (param.type_name.as_str(), arg.is_empty()) {
+			("Option", true) => "None".to_string(), /* If the argument is Option and empty, */
+			// replace it with `None`
+			("Option", false) => format!("Some({})", arg), /* If the argument is Option and not */
+			// empty, wrap it in `Some(...)`
+			_ => arg, // If the argument is not Option, return it as is
+		})
+		.collect::<Vec<String>>())
+}
+
+/// Processes a list of argument values for a specified contract constructor,
+/// wrapping each value in `Some(...)` or replacing it with `None` if the argument is optional
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+/// * `constructor` - Label of the contract constructor to retrieve.
+/// * `args` - Argument values provided by the user.
+pub fn process_constructor_args<P>(
```

Comment:
This function does almost the same than `process_message_args`.

Reply:
I initially planned to leave it as-is, but I agree it's a good idea to improve it now. On it.